radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -14,10 +14,25 @@ use radix_engine_interface::data::{IndexedScryptoValue, ScryptoDecode};
 use transaction::manifest::decompiler::DecompilationContext;
 use utils::ContextualDisplay;
 
+#[cfg(not(feature = "resource-usage-with-cpu"))]
+#[derive(Debug, Clone, Default, ScryptoEncode, ScryptoDecode)]
+pub struct ResourcesUsage {
+    pub heap_allocations_sum: usize,
+    pub heap_peak_memory: usize,
+}
+#[cfg(feature = "resource-usage-with-cpu")]
+#[derive(Debug, Clone, Default, ScryptoEncode, ScryptoDecode)]
+pub struct ResourcesUsage {
+    pub heap_allocations_sum: usize,
+    pub heap_peak_memory: usize,
+    pub cpu_cycles: u64,
+}
+
 #[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
 pub struct TransactionExecution {
     pub fee_summary: FeeSummary,
     pub events: Vec<TrackedEvent>,
+    pub resources_usage: ResourcesUsage,
```

Comment:
So, we're puting resource usage under the `execution` section of the `TransactionReceipt`, instead of the `result` section.

This is good as I suppose node is not merkelizing `execution`.

Reply:
> This is good as I suppose node is not merkelizing execution.
What do you mean by that?

Reply:
I mean the `execution` is not agreed by validators through consensus.